### PR TITLE
Merge partition_full_outer_join back to partition1.

### PR DESCRIPTION
**Pull request by xinzweb**

### CODE REVIEW

**Comment by vraghavan78** [comment]
+1

File: src/test/regress/expected/partition1.out [hunk 1]
```diff
@@ -2308,10 +2308,10 @@ insert into d values(1, 1, 2);
...
 insert into d values(1, 100, 2000);
-ERROR:  no partition for partitioning key  (seg0 slarimac:40000 pid=97859)
+ERROR:  no partition for partitioning key  (seg0 xzhangmac:40000 pid=75483)
 insert into d values(1, '1000', '1001'), (1, '1001', '1002'), (1, '1003', '1004');
 insert into d values(1, 100, NULL);
-ERROR:  no partition for partitioning key  (seg0 slarimac:40000 pid=97859)
+ERROR:  no partition for partitioning key  (seg0 xzhangmac:40000 pid=75483)
 select * from d_1_prt_a;
  a | b | c 
 ---+---+---
```
Comment (hsyuan) [inline src/test/regress/expected/partition1.out]: See line 2382:
```
 create table b (i int, n numeric(20, 2), t timestamp, s text)
 distributed by (i)
 partition by range(n, t, s)
 (
 start(2000.99, '2007-01-01 00:00:00', 'AAA')
   end (4000.95, '2007-02-02 15:00:00', 'BBB'),
 start(2000.99, '2007-01-01 00:00:00', 'BBB')
   end (4000.95, '2007-02-02 16:00:00', 'CCC'),
 start(4000.95, '2007-01-01 00:00:00', 'AAA')
   end (7000.95, '2007-02-02 15:00:00', 'BBB')
 );
 ERROR:  too many columns for RANGE partition -- only one column is allowed.
 LINE 3: partition by range(n, t, s)
                      ^
 -- should work
 insert into b values(1, 2000.99, '2007-01-01 00:00:00', 'AAA');
 ERROR:  relation "b" does not exist
 insert into b values(2, 2000.99, '2007-01-01 00:00:00', 'BBB');
 ERROR:  relation "b" does not exist
 insert into b values(3, 4000.95, '2007-01-01 00:00:00', 'AAA');
 ERROR:  relation "b" does not exist
 insert into b values(6, 3000, '2007-02-02 15:30:00', 'BBC');
 ERROR:  relation "b" does not exist
 insert into b values(6, 3000, '2007-02-02 15:30:00', 'CC');
 ERROR:  relation "b" does not exist
 insert into b values(6, 3000, '2007-02-02 16:00:00'::timestamp - 
 					'1 second'::interval, 'BBZZZZZZZZZZ');
```

The query that is expected to work did not work. The table b is not even created, making the test meaningless.

Comment (xinzweb) [inline src/test/regress/expected/partition1.out]: Good spot!

@hlinnaka do you happen to know whether GPDB support 'multi value range' partitioning or not? If not, then maybe just the 'negative' test for 'create table b' would be good enough?

Comment (hlinnaka) [inline src/test/regress/expected/partition1.out]: On 14/01/16 22:14, Xin Zhang wrote:
> @hlinnaka do you happen to know whether GPDB support 'multi value
> range' partitioning or not? If not, then maybe just the 'negative'
> test for 'create table b' would be good enough?

Sorry, I have no clue. What is "multi value range" partitioning?

- Heikki

Comment (hsyuan) [inline src/test/regress/expected/partition1.out]: I think the following example is the right method to write multi range partition:
```sql
CREATE TABLE p3_sales (id int, year int, month int, day int, 
region text)
DISTRIBUTED BY (id)
PARTITION BY RANGE (year)
    SUBPARTITION BY RANGE (month)
       SUBPARTITION TEMPLATE (
        START (1) END (13) EVERY (1), 
        DEFAULT SUBPARTITION other_months )
( START (2002) END (2012) EVERY (1), 
  DEFAULT PARTITION outlying_years );
```

Only one column is allowed for RANGE partition. But in this test case, it has 3 columns for range partition, which is wrong in syntax.

**Pull request closed without merging**